src/setfit/data.py

```diff
@@ -110,21 +156,24 @@ def create_fewshot_splits(
     dataset: Dataset,
     sample_sizes: List[int],
     add_data_augmentation: bool = False,
-    dataset_name: Optional[str] = None,
+    dataset_name: str = None,
```

Comment:
```suggestion
    dataset_name: Optional[str] = None,
```
[PEP 484](https://peps.python.org/pep-0484/#union-types) again, if `None` is the default, then the type hint must include `Optional`.